src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -233,31 +281,35 @@ public void handleMessage(Message msg) {
 
                 try {
                     int returnCode = MPDbAdapter.DB_UNDEFINED_CODE;
+                    String token = null;
 
                     if (msg.what == ENQUEUE_PEOPLE) {
-                        final JSONObject message = (JSONObject) msg.obj;
+                        final PeopleDescription message = (PeopleDescription) msg.obj;
 
                         logAboutMessageToMixpanel("Queuing people record for sending later");
                         logAboutMessageToMixpanel("    " + message.toString());
-
-                        returnCode = mDbAdapter.addJSON(message, MPDbAdapter.Table.PEOPLE);
+                        token = message.getToken();
+                        returnCode = mDbAdapter.addJSON(message.getMessage(), token, MPDbAdapter.Table.PEOPLE, false);
                     } else if (msg.what == ENQUEUE_EVENTS) {
                         final EventDescription eventDescription = (EventDescription) msg.obj;
                         try {
                             final JSONObject message = prepareEventObject(eventDescription);
                             logAboutMessageToMixpanel("Queuing event for sending later");
                             logAboutMessageToMixpanel("    " + message.toString());
-                            returnCode = mDbAdapter.addJSON(message, MPDbAdapter.Table.EVENTS);
+                            token = eventDescription.getToken();
+                            returnCode = mDbAdapter.addJSON(message, token, MPDbAdapter.Table.EVENTS, eventDescription.isAutomatic());
                         } catch (final JSONException e) {
                             MPLog.e(LOGTAG, "Exception tracking event " + eventDescription.getEventName(), e);
                         }
                     } else if (msg.what == FLUSH_QUEUE) {
                         logAboutMessageToMixpanel("Flushing queue due to scheduled or forced flush");
                         updateFlushFrequency();
-                        sendAllData(mDbAdapter);
-                        if (SystemClock.elapsedRealtime() >= mDecideRetryAfter) {
+                        token = (String) msg.obj;
```

Comment:
that's a good point. the problem is that we use methods like `hasMessages` and `removeMessage` in other parts of the code so we'd need to keep a reference to the object that we pass on this message. Since the token (an string) is passed on every message (PeopleDescription, EventDescription, etc..) it makes it easier to keep the string as the object so we can remove those messages in the queue.